src/Common/MSBuildCachePluginBase.cs

```diff
@@ -354,6 +355,15 @@ private async Task<CacheResult> GetCacheResultInnerAsync(BuildRequestData buildR
         CheckForDuplicateOutputs(logger, nodeBuildResult.Outputs, nodeContext);
 
         await FinishNodeAsync(logger, nodeContext, pathSet, nodeBuildResult);
+        // In the case of a build-and-test request, we want to cache the build results but and the test execution (i.e, only execute tests on cache misses).
+        _ = bool.TryParse(projectInstance.GetPropertyValue("IsTestProject"), out bool isTestProject);
+        if (isBuildAndTestRequest && isTestProject)
+        {
+            // this property can be read by downstream tasks, such as a Test Target, to determine if the test execution should be skipped
+            _ = projectInstance.SetProperty("SkipExecution", "true");
```

Comment:
This seems dangerous for the cache plugin to mutate the project instance. I'm actually not even sure it even works in practice considering this is happening in the scheduler node and not the worker node.

Reply:
This works from my testing.

Reply:
Also, this is a property of the runvstest task/sdk not the test project